services/storage/src/simcore_service_storage/simcore_s3_dsm.py

```diff
@@ -203,6 +182,26 @@ async def list_files(  # noqa C901
                     for node_id, node_data in proj_data.workbench.items()
                 }
 
+        # expand directories until the max number of files to return is not reached
+        directory_expands: list[Coroutine] = []
+        for metadata in file_and_directory_meta_data:
+            if (
+                expand_dirs
+                and metadata.is_directory
+                and len(data) < EXPAND_DIR_MAX_ITEM_COUNT
+            ):
+                max_items_to_include = EXPAND_DIR_MAX_ITEM_COUNT - len(data)
+                directory_expands.append(
+                    expand_directory(
+                        self.app,
+                        self.simcore_bucket_name,
+                        metadata,
+                        max_items_to_include,
+                    )
+                )
+        for files_in_directory in await logged_gather(*directory_expands):
```

Comment:
no need to limit the number of tasks here right? just that now this does not overwhelm S3/CEPH

Reply:
Let me put a limit for good measure. forgot about it, thanks